Exercise_2.java

```diff
@@ -8,18 +9,31 @@
   
         StackNode(int data) 
         { 
-            //Constructor here 
+        	next.data = data;
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
+    	if(root == null)
+    		return false;

```

Comment:
if root is null, that implies the stack is empty. Shouldn't it be returning true here and false otherwise?